### PR TITLE
Test remaining transitions api examples

### PR DESCRIPTION
**Pull request by derdilla**
Adds tests for `relative_positioned_transition`, `positioned_transition`, `sliver_fade_transition`, `align_transition`, `animated_builder`, `rotation_transition`, `animated_widget`, `slide_transition`, `listenable_builder`, `scale_transition`, `default_text_style_transition`, `decorated_box_transition`, `size_transition` api examples. Makes double type in the `align_transition` example explicit.

A test for `fade_transition` is already in currently open #148178.

Part of #130459.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

File: examples/api/test/widgets/transitions/align_transition.0_test.dart [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import 'package:flutter/material.dart';
+import 'package:flutter_api_samples/widgets/transitions/align_transition.0.dart' as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('Shows flutter logo in transition', (WidgetTester tester) async {
+    await tester.pumpWidget(const example.AlignTransitionExampleApp());
+    expect(find.byType(ColoredBox), findsOneWidget);
+    expect(find.byWidgetPredicate((Widget padding) => padding is Padding
+        && padding.padding == const EdgeInsets.all(8.0)), findsOneWidget);
```
Comment (TahaTesser) [inline examples/api/test/widgets/transitions/align_transition.0_test.dart:14]:
```suggestion
    expect(
      find.byWidgetPredicate((Widget padding) => padding is Padding
        && padding.padding == const EdgeInsets.all(8.0)),
      findsOneWidget,
    );
```

File: examples/api/test/widgets/transitions/align_transition.0_test.dart [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        && padding.padding == const EdgeInsets.all(8.0));
+
+    expect(
+      tester.getBottomLeft(paddingFinder),
+      tester.getBottomLeft(find.byType(AlignTransition)),
+    );
+
+    await tester.pump();
+    await tester.pump(const Duration(seconds: 2));
+    await tester.pump();
```
Comment (TahaTesser) [inline examples/api/test/widgets/transitions/align_transition.0_test.dart:32]:
```suggestion
    await tester.pump(const Duration(seconds: 2));
    await tester.pump();
```

File: examples/api/test/widgets/transitions/align_transition.0_test.dart [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    expect(
+      tester.getCenter(paddingFinder),
+      tester.getCenter(find.byType(AlignTransition)),
+    );
+
+    await tester.pump();
+    await tester.pump(const Duration(seconds: 2));
+    await tester.pump();
```
Comment (TahaTesser) [inline examples/api/test/widgets/transitions/align_transition.0_test.dart:41]:
```suggestion
    await tester.pump(const Duration(seconds: 2));
    await tester.pump();
```

File: examples/api/test/widgets/transitions/animated_builder.0_test.dart [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    expect(tester.widget(find.byType(Container)), isA<Container>()
+      .having((Container container) => container.color, 'color', Colors.green));
+
+    expect(find.byWidgetPredicate((Widget widget) => widget is Transform
+        && widget.transform == Transform.rotate(angle: 0.0).transform),
+      findsOneWidget);
+
+    await tester.pump();
+    await tester.pump(const Duration(seconds: 5));
+    await tester.pump();
```
Comment (TahaTesser) [inline examples/api/test/widgets/transitions/animated_builder.0_test.dart:25]:
```suggestion
    await tester.pump(const Duration(seconds: 5));
    await tester.pump();
```

File: examples/api/test/widgets/transitions/scale_transition.0_test.dart [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+      of: find.byType(Center),
+      matching: find.byType(Transform)
+    );
+
+    Transform transform = tester.widget(transformFinder);
+    expect(transform.transform[0], 0);
+
+    await tester.pump();
+    await tester.pump(const Duration(seconds: 2));
+    await tester.pump();
```
Comment (TahaTesser) [inline examples/api/test/widgets/transitions/scale_transition.0_test.dart:37]:
```suggestion
    await tester.pump(const Duration(seconds: 2));
    await tester.pump();
```

Please remove the initial pump as it's not necessary here and from other tests

**Review by TahaTesser — CHANGES_REQUESTED** [review]

File: examples/api/test/widgets/transitions/size_transition.0_test.dart [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+    expect(
+      tester.widget(find.byType(SizeTransition)),
+      isA<SizeTransition>()
+        .having((SizeTransition transition) => transition.axis, 'axis', Axis.horizontal)
+        .having((SizeTransition transition) => transition.axisAlignment, 'axis alignment', -1)
+        .having((SizeTransition transition) => transition.sizeFactor, 'factor', isA<CurvedAnimation>()
+          .having((CurvedAnimation animation) => animation.curve, 'curve', Curves.fastOutSlowIn)
+          .having((CurvedAnimation animation) => animation.parent, 'paren', isA<AnimationController>()
+            .having((AnimationController controller) => controller.duration, 'duration', const Duration(seconds: 3)))
```
Comment (TahaTesser) [inline examples/api/test/widgets/transitions/size_transition.0_test.dart:28]: Please align these lines.

File: examples/api/test/widgets/transitions/listenable_builder.2_test.dart [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    await tester.pump();
+    expect(find.text('0'), findsNothing);
+    expect(find.text('1'), findsOneWidget);
+
+    await tester.tap(find.byIcon(Icons.add));
+    await tester.tap(find.byIcon(Icons.add));
+    await tester.tap(find.byIcon(Icons.add));
+    await tester.tap(find.byIcon(Icons.add));
+    await tester.pump();
+    expect(find.text('5'), findsOneWidget);
```
Comment (TahaTesser) [inline examples/api/test/widgets/transitions/listenable_builder.2_test.dart:33]:
```suggestion
    await tester.tap(find.byIcon(Icons.add));
    await tester.tap(find.byIcon(Icons.add));
    await tester.tap(find.byIcon(Icons.add));
    await tester.tap(find.byIcon(Icons.add));
    await tester.pump();
    expect(find.text('5'), findsOneWidget);
```

Use a `for` loop if you want to repeat the action

File: examples/api/test/widgets/transitions/sliver_fade_transition.0_test.dart [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+    expect(find.byType(SliverFadeTransition), findsOneWidget);
+    expect(find.byType(CustomScrollView), findsOneWidget);
+    expect(find.byWidgetPredicate(
+      (Widget widget) => widget is Container &&
+        widget.color == Colors.indigo[200]
+    ), findsNWidgets(3));
+    expect(find.byWidgetPredicate(
+            (Widget widget) => widget is Container &&
+            widget.color == Colors.orange[200]
+    ), findsNWidgets(2));
```
Comment (TahaTesser) [inline examples/api/test/widgets/transitions/sliver_fade_transition.0_test.dart:23]: Too much leading indent

**Review by TahaTesser — CHANGES_REQUESTED** [review]

File: examples/api/test/widgets/transitions/size_transition.0_test.dart (hunk 6 above)
Comment (NobodyForNothing) [reply examples/api/test/widgets/transitions/size_transition.0_test.dart:28]: Just to confirm: Does this apply when they are nested?

File: examples/api/test/widgets/transitions/sliver_fade_transition.0_test.dart (hunk 8 above)
Comment (NobodyForNothing) [reply examples/api/test/widgets/transitions/sliver_fade_transition.0_test.dart]: Thanks

**Comment by NobodyForNothing** [comment]
Thanks for the reviews so far. FYI: I will be on vacation over the next ~2 weeks and will fix any issues that come up during that time after that.

**Review by TahaTesser — APPROVED** [review]
LGTM!

**Review by bleroux — APPROVED** [review]
LGTM! 13 new tests :+1:

**Pull request merged**